src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/VsUtilities.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    internal static class VsUtilities
+    {
+        public static void ApplyTextEdits(
+            IServiceProvider serviceProvider,
+            Uri uri,
+            ITextSnapshot snapshot,
+            IEnumerable<TextEdit> textEdits)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
If this was a service you could use joinable task context here and you could avoid the need to pass in the `serviceProvider`